src/apps/vaporwave-finance/aurora/vaporwave-finance.vault.token-fetcher.ts

```diff
@@ -0,0 +1,201 @@
+import { Register } from '~app-toolkit/decorators';
+import { PositionFetcher } from '~position/position-fetcher.interface';
+import { AppTokenPosition } from '~position/position.interface';
+import { Network } from '~types/network.interface';
+import { ContractType } from '~position/contract.interface';
+
+import { VaporwaveFinanceContractFactory } from '../contracts';
+import { VAPORWAVE_FINANCE_DEFINITION } from '../vaporwave-finance.definition';
+import { buildDollarDisplayItem } from "~app-toolkit/helpers/presentation/display-item.present";
+import { BaseToken } from '~position/token.interface';
+
+
+const appId = VAPORWAVE_FINANCE_DEFINITION.id;
+const groupId = VAPORWAVE_FINANCE_DEFINITION.groups.vault.id;
+const network = Network.AURORA_MAINNET;
+
+
+export type VaporwaveVaultDetails = {
+  id: string
+  name: string // label
+  earnContractAddress: string // vault address
+  earnedToken: string // vault token
+
+  oracleId: string // NOT a proper symbol
+  tokenAddress: string // want
+  tokenDecimals: number // decimal
+
+  pricePerFullShare: number // ratio
+
+  logo: string
+  status: string // check for retired
+  assets: string[]
+};
+
+
+export async function getBaseERC20Token(
+  address: string,
+  appToolkit: IAppToolkit): Promise<BaseToken> {
+
+  const multicall = appToolkit.getMulticall(network);
+  const tokenContract = appToolkit.globalContracts.erc20({ address: address, network });
+  const [symbol, decimal] = await Promise.all([
+    multicall
+      .wrap(tokenContract)
+      .symbol()
+      .catch(() => ""),
+    multicall
+      .wrap(tokenContract)
+      .decimals()
+      .catch(() => 0),
+  ]);
+
+  const token: BaseToken = {
+    type: ContractType.BASE_TOKEN,
+    address: address,
+    network: network,
+    price: 0,
+    symbol: symbol,
+    decimals: decimal,
+  };
+
+  return token
+
+}
+
+@Register.TokenPositionFetcher({ appId, groupId, network })
+export class AuroraVaporwaveFinanceVaultTokenFetcher implements PositionFetcher<AppTokenPosition> {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(VaporwaveFinanceContractFactory)
+    private readonly vaporwaveFinanceContractFactory: VaporwaveFinanceContractFactory,
+  ) { }
+
+  async getPositions() {
+    // http://localhost:5001/apps/vaporwave-finance/tokens?groupIds[]=vault&network=aurora
+    const vaultData = await Axios.get<VaporwaveVaultDetails[]>("https://api.vaporwave.farm/vaults").then(
+      (v) => v.data
+    );
+    const vtokenPrices = await Axios.get("https://api.vaporwave.farm/vaportokenprices").then(
+      (v) => v.data[network]
+    );
+    const apyData = await Axios.get("https://api.vaporwave.farm/apy").then(
+      (v) => v.data
+    )
+    const wantPrices = await Axios.get("https://api.vaporwave.farm/lps").then(
+      (v) => v.data
+    )
+    const baseTokenPrices = await Axios.get("https://api.vaporwave.farm/prices").then(
+      (v) => v.data
+    )
```

Comment:
Don't use prices for the Vaporwave API. This makes it such that the prices for tokens end up not reflecting other prices in our application. Instead, use the Zapper prices as follows:

```ts
// Get base token prices (USDT, VWAVE, etc.)
const baseTokens = await this.appToolkit.getBaseTokenPrices(network);
// Get dependency app token prices (i.e: Trisolaris, Rose, etc... we're missing these)
const appTokens = await this.appToolkit.getAppTokenPositions(...dependencies);

// ...

// Find the match
const wantToken = [...appTokens, ...baseTokens].find(v => v.address === vault.tokenAddress)
```